File: jee6RestScala/src/main/scala/org/timoponce/Persister.scala
package org.timoponce

import javax.persistence.{PersistenceContext, EntityManager}
import java.util

class Persister {

  @PersistenceContext
  var entityManager: EntityManager = _

  def persist(t: AnyRef) = {
    entityManager.persist(t)
  }

  def getAll[T]( targetClass: Class[T] ): util.List[T]={
    val className = targetClass.getSimpleName
    val result = entityManager.createQuery("SELECT t FROM "+ className +" t").getResultList
    return result.asInstanceOf[ util.List[T] ]
  }

}

File: jee6Webapp/src/main/java/org/timoponce/Persister.java
package org.timoponce;

import javax.ejb.*;
import javax.persistence.*;
import java.util.*;
import javax.transaction.UserTransaction;

@Stateless @LocalBean
public class Persister {

  @PersistenceContext
  EntityManager entityManager;

  	public <T>void persist(T t) {
		entityManager.persist(t);	
  	}

	public <T>List<T> list(Class<T> target){
		return (List<T>)entityManager
			.createQuery("SELECT t FROM " + target.getSimpleName() + " t")
			.getResultList();
	}
}
